src/main/java/fr/insee/pogues/service/VariableService.java

```diff
@@ -82,9 +85,17 @@ private List<VariableType> getQuestionnaireVariables(Questionnaire questionnaire
      */
     private void computeScopeNameFromScopeId(VariableType variable, Questionnaire.Iterations iterations) {
         String scopeId = variable.getScope();
-        if (scopeId != null) {
-            IterationType iteration = iterations.getIteration().stream().filter(v -> scopeId.equals(v.getId())).toList().getFirst();
-            if (iteration != null) variable.setScope(iteration.getName());
+        if (scopeId == null) return;
+
+        Optional<IterationType> iteration = iterations.getIteration().stream().filter(v -> {
```

Comment:
if iterations is null -> 500

iterations is null when there is no scope (only root scope = "Questionnaire")